specification/schemaregistry/data-plane/Microsoft.EventHub/preview/2018-01-01-preview/schemaregistry.json

```diff
@@ -1,132 +1,224 @@
 {
   "swagger": "2.0",
   "info": {
-    "version": "2018-01-01-preview",
-    "title": "schemaregistry",
-    "description": "Description of the new service",
-    "x-ms-code-generation-settings": {
-      "name": "schemaregistryClient"
-    }
-  },
-  "host": "management.azure.com",
-  "schemes": ["https"],
-  "consumes": ["application/json"],
-  "produces": ["application/json"],
-  "security": [
-    {
-      "azure_auth": ["user_impersonation"]
-    }
-  ],
-  "securityDefinitions": {
-    "azure_auth": {
-      "type": "oauth2",
-      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
-      "flow": "implicit",
-      "description": "Azure Active Directory OAuth2 Flow",
-      "scopes": {
-        "user_impersonation": "impersonate your user account"
-      }
-    }
+    "title": "Azure Schema Registry",
+    "version": "2018-01-01-preview"
   },
   "paths": {
-    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.EventHub/operations": {
+    "/$schemagroups/getSchemaById/{schema-id}": {
       "get": {
-        "tags": ["Tag1"],
-        "operationId": "OperationGroup_Get",
+        "summary": "Get schema by ID",
+        "description": "Get schema by ID.  Azure Schema Registry guarantees that ID is unique within a namespace.",
+        "x-ms-examples": {
+          "Get schema by ID": {
+            "$ref": "./examples/getSchemaById.json"
+          }
+        },
+        "operationId": "getSchemaById",
```

Comment:
Usually the pattern for operation IDs is NOUN_VERB.  Most code generators use the NOUN to generate the name of the containing class/interface, and VERB for the method name within that class/interface.  By omitting the NOUN you get whatever default value the generators use which varies per language.

I recommend something like `Schema_GetById`.

Reply:
Also keep in mind that different NOUNs generate different classes/interfaces.  So if you want all of these methods to belong to the same class/interface, they should have the same NOUN.

Reply:
Converted all to Schema_OperationName